src/errorBoundary.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { ReactNode } from "react";
+
+interface ErrorBoundaryProps {
+  children: ReactNode;
+}
+
+interface ErrorBoundaryState {
+  hasError: boolean;
+  error: { message: string; stack: string };
+}
+
+class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  constructor(props: ErrorBoundaryProps) {
+    super(props);
+    this.state = { hasError: false, error: { message: "", stack: "" } };
+  }
+
+  static getDerivedStateFromError(error: ErrorBoundaryState["error"]): ErrorBoundaryState {
+    // Update state so the next render will show the fallback UI.
+    return { hasError: true, error };
+  }
+
+  render(): ReactNode {
+    if (this.state.hasError) {
+      // You can render any custom fallback UI
+      return (
+        <div>
+          <h1 className="text-xl font-bold mb-2">Document could not be rendered</h1>
+          <div className="mb-2">
+            <span className="font-medium">Message:</span> {this.state.error.message}
+          </div>
+          <div className="mb-2">
+            <span className="font-medium">Stack:</span> {this.state.error.stack}
+          </div>
+        </div>
+      );
+    }
+
+    return this.props.children;
+  }
+}
+
+export default ErrorBoundary;
```

Comment:
add a storybook maybe something like this -> https://github.com/TradeTrust/document-creator-website/blob/master/src/components/ErrorBoundary/ErrorBoundary.stories.tsx

Reply:
ok